posttroll/publisher.py

```diff
@@ -262,3 +262,41 @@ def __enter__(self):
 
     def __exit__(self, exc_type, exc_val, exc_tb):
         return self.stop()
+
+
+class PublisherContainer(object):
+
+    """Threaded container for a publisher instance"""
+
+    logger = logging.getLogger("PublisherContainer")
+
+    def __init__(self, name, port=0, aliases=None, broadcast_interval=2,
+                 nameservers=None):
+        self.pub = None
+        self.create_publisher(name, port=port, aliases=aliases,
+                              broadcast_interval=broadcast_interval,
+                              nameservers=nameservers)
+        self.thread = Thread(target=self.pub.start)
+        self.thread.setDaemon(True)
+        self.thread.start()
+
+    def create_publisher(self, name, port=0, aliases=None,
+                         broadcast_interval=2,
+                         nameservers=None):
+        """Create a publisher instance"""
+        if self.pub is None:
+            self.pub = NoisyPublisher(name, port=port, aliases=aliases,
+                                      broadcast_interval=broadcast_interval,
+                                      nameservers=nameservers)
+
+    def send(self, msg):
+        """Send message"""
+        self.pub.send(str(msg))
+        self.logger.info("Sent message: %s", str(msg))
```

Comment:
Is this thread safe ? what's the difference then with NoisyPublisher ? Because there is no `run` method implemented...